aws/resource_aws_emr_cluster.go

```diff
@@ -485,6 +504,181 @@ func resourceAwsEMRCluster() *schema.Resource {
 	}
 }
 
+func InstanceFleetConfigSchema() *schema.Resource {
+	return &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"instance_fleet_type": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringInSlice(emr.InstanceFleetType_Values(), false),
+			},
```

Comment:
This should be automatically set by whether the `master_instance_fleet` or `core_instance_fleet` configuration block is being used.

```suggestion
```

Reply:
@bflad  The reason I have `instance_fleet_type` here is to be able to reuse the `InstanceFleetConfigSchema` 
and related function between master_fleet, core_fleet and task_fleet.
Let me see if there is a better way to do it. I just don't want to copy paste the code over.